simple_functorch.py

```diff
@@ -510,6 +515,25 @@ def propagate(dL_doutputs: List[Tensor]):
         )
         return r, saved
 
+    def custom_vmap(self, fn, batch_rule, *args):
```

Comment:
This appears to diverge substantially from JAX's custom_vmap, at https://github.com/google/jax/blob/main/jax/_src/custom_batching.py

Reply:
@ezyang is your comment that the API is different, the implementation is different, or both?

Reply:
implementation. But later I worked out that this is exactly analogous to how we did batching, so... idk, maybe it's still fine